pkg/kapis/devops/v1alpha2/devops.go

```diff
@@ -318,65 +317,36 @@ func (h *ProjectPipelineHandler) GetPipelineRunNodes(req *restful.Request, resp
-// 2. no particular submitters
-// only the owner of this Pipeline can approve or reject it
+// approvableCheck requires the users who have PipelineRun management permission to
+// approve a PipelineRun. If the particular submitters exist, we also restrict the users
+// who are in the particular submitters.
 func (h *ProjectPipelineHandler) approvableCheck(nodes []clientDevOps.NodesDetail, pipe pipelineParam) {
-	var userInfo user.Info
-	var ok bool
-	var isAdmin bool
-	// check if current user belong to the admin group, grant it if it's true
-	if userInfo, ok = request.UserFrom(pipe.Context); ok {
-		//createAuth := authorizer.AttributesRecord{
-		//	User:            userInfo,
-		//	Verb:            authorizer.VerbDelete,
-		//	Workspace:       pipe.Workspace,
-		//	DevOps:          pipe.ProjectName,
-		//	Resource:        "devopsprojects",
-		//	ResourceRequest: true,
-		//	ResourceScope:   request.DevOpsScope,
-		//}
-		//
-		//if decision, _, err := h.authorizer.Authorize(createAuth); err == nil {
-		//	isAdmin = decision == authorizer.DecisionAllow
-		//} else {
-		//	// this is an expected case, printing the debug info for troubleshooting
-		//	klog.V(8).Infof("authorize failed with '%v', error is '%v'",
-		//		createAuth, err)
-		//}
-	} else {
+	userInfo, ok := request.UserFrom(pipe.Context)
+	if !ok {
 		klog.V(6).Infof("cannot get the current user when checking the approvable with pipeline '%s/%s'",
 			pipe.ProjectName, pipe.Name)
 		return
 	}
 
-	var createdByCurrentUser bool // indicate if the current user is the owner
-	if pipeline, err := h.devopsOperator.GetPipelineObj(pipe.ProjectName, pipe.Name); err == nil {
-		if creator, ok := pipeline.GetAnnotations()[constants.CreatorAnnotationKey]; ok {
-			createdByCurrentUser = userInfo.GetName() == creator
-		} else {
-			klog.V(6).Infof("annotation '%s' is necessary but it is missing from '%s/%s'",
-				constants.CreatorAnnotationKey, pipe.ProjectName, pipe.Name)
-		}
-	} else {
-		klog.V(6).Infof("cannot find pipeline '%s/%s', error is '%v'", pipe.ProjectName, pipe.Name, err)
-		return
-	}
-
 	// check every input steps if it's approvable
-	for i, node := range nodes {
+	for i := range nodes {
+		node := &nodes[i]
 		if node.State != clientDevOps.StatePaused {
 			continue
 		}
 
-		for j, step := range node.Steps {
+		for j := range node.Steps {
+			step := &node.Steps[j]
 			if step.State != clientDevOps.StatePaused || step.Input == nil {
 				continue
 			}
-
-			nodes[i].Steps[j].Approvable = isAdmin || createdByCurrentUser || step.Input.Approvable(userInfo.GetName())
+			if len(step.Input.GetSubmitters()) == 0 {
+				// TODO: no particular submitters, we make the PipelineRun approvable by default.
+				// Until we can obtain the permissions of the currently logged in user
+				step.Approvable = true
+			} else {
+				step.Approvable = step.Input.Approvable(userInfo.GetName())
```

Comment:
What happens if there is a user A who is the owner of this Pipeline, and A is not a submitter. In this case, A will not be able to approve or reject it.

IMO, it is weird.